tools/gyp/pylib/gyp/xcodeproj_file.py

```diff
@@ -788,8 +784,7 @@ def UpdateProperties(self, properties, do_copy=False):
             self._properties[property] = value.Copy()
           else:
             self._properties[property] = value
-        elif isinstance(value, str) or isinstance(value, unicode) or \
-             isinstance(value, int):
+        elif isinstance(value, (basestring, int)):
```

Comment:
`string_types + (int,)`? (Assuming it's a tuple and not a list.)

Reply:
It is a tuple. https://docs.python.org/3/library/functions.html#isinstance

Reply:
Okay, good to know :-). But my question really was "why `basestring` instead of `string_types`?" since that's what you're using elsewhere.

Reply:
In this file (lines 147-152) we redefine __basestring__, __cmp__, and __unicode__ which were all dropped from Python 3 therefore we do not need to __from gyp import string_types__.

Reply:
Right, now I understand. Objection withdrawn then.